gossip/src/cluster_info.rs

```diff
@@ -1972,6 +1972,10 @@ impl ClusterInfo {
     // Returns a predicate checking if the pull request is from a valid
     // address, and if the address have responded to a ping request. Also
     // appends ping packets for the addresses which need to be (re)verified.
+    //
+    // allow lint false positive trait bound requirement (`CryptoRng` only
+    // implemented on `&'a mut T`
+    #[rustversion::attr(since(1.73), allow(clippy::needless_pass_by_ref_mut))]
```

Comment:
comment pretty much covers it. allegedly this lint was introduced in 1.72, but that version was complaining that it doesn't know about it, hence the toolchain version gating.

i'm guessing this one won't be contentious and we can just put it in with a corresponding tracking issue